src/main/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepository.java

```diff
@@ -251,7 +252,7 @@ protected Flux mapFlux(Observable observable) {
         return mapMono(operations.queryView(query)
                 .map(AsyncViewResult::rows)
                 .flatMap(row -> {
-                    AsyncViewRow asyncViewRow = (AsyncViewRow) row;
+                    AsyncViewRow asyncViewRow = row.toBlocking().single();
```

Comment:
That's a blocking call within a non-blocking flow. It should be rather flat-mapped in the operator before instead enforcing synchronization here.